ext/ddtrace_profiling_native_extension/collectors_stack.c

```diff
@@ -80,20 +112,88 @@ static VALUE _native_sample(DDTRACE_UNUSED VALUE _self, VALUE thread, VALUE reco
 
   sampling_buffer *buffer = sampling_buffer_new(max_frames_requested);
 
-  sample_thread(
-    thread,
-    buffer,
-    recorder_instance,
-    (ddog_Slice_i64) {.ptr = metric_values, .len = ENABLED_VALUE_TYPES_COUNT},
-    (ddog_Slice_label) {.ptr = labels, .len = labels_count}
-  );
+  if (in_gc == Qnil || in_gc == Qfalse) {
```

Comment:
```suggestion
  if (!RB_TEST(in_gc)) {
```
https://github.com/ruby/ruby/blob/7837dccb0ea9f84f033635e77bc9dfb6ea023bcd/include/ruby/internal/special_consts.h#L123-L149

Reply:
I'll admit I thought "wait I think there was a better way of doing it..." and I did search a bit but I didn't find it. Thanks a lot for the reminder! :)